File: solutions/typescript/nth-prime/1/nth-prime.ts
export function nth(n: number): number {
  if (n < 1) {
    throw 'Prime is not possible'
  }
  let p = 2
  let primes: number[] = []
  while (primes.length < n) {
    primes.push(p)
    if (primes.length == n) {
      break
    }
    p = p + 1
    while (primes.some(d => p % d == 0)) {
      p += 1
    }
  }
  return p
}

File: solutions/csharp/nth-prime/1/NthPrime.cs
public static class NthPrime
{
    public static int Prime(int nth)
    {
        var primes = new List<int>();
        var p = 2;
        for (var i = 0; i < nth; i++)
        {
            while (primes.Any(q => p % q == 0))
            {
                p++;
            }
            primes.Add(p++);
        }
        return p - 1;
    }
}